src/ixdat/spectra.py

```diff
@@ -505,9 +522,31 @@ def from_spectrum_list(cls, spectrum_list, **kwargs):
         obj_as_dict["field"] = field
         obj_as_dict["technique"] = technique
         del obj_as_dict["field_id"]
-        obj_as_dict.update(kwargs)
+        # Any attribute we want in the SpecrumSeries for each spectrum should go here:
+        obj_as_dict["durations"] = [s.duration for s in spectrum_list]
         return cls.from_dict(obj_as_dict)
 
+    @property
+    def field(self):
+        """Since a spectrum can be loaded lazily, we make sure the field is loaded
+
+        We also want to make sure that the field has the tstamp of the SpectrumSeries.
+        """
+        if isinstance(self._field, PlaceHolderObject):
+            self._field = self._field.get_object()
+
+        if abs(self._field.axes_series[0].tstamp - self.tstamp) > self.t_tolerance:
+            self._field = Field(
+                name=self._field.name,
+                data=self._field.data,
+                unit_name=self._field.unit_name,
+                axes_series=[
+                    time_shifted(self._field.axes_series[0], tstamp=self.tstamp),
+                    self._field.axes_series[1],
+                ],
+            )
```

Comment:
Not sure I completely understand. I think the `t_tolerance` should not change when `tstamp` is changed. `t_tolerance` is related to the time spacing between the spectra, which doesn't change. `tolerance_by_spacing`, to which `t_tolerance` defaults, is likewise unchanged, since it is the time spanned by the `tseries` (i.e., `t[-1] - t[0]`), which doesn't change when `t` is shifted by a constant.